README.md
# Advanced Custom Fields (ACF): Animate.css Effects Field

Contributors: Thomas A. Reinert, TAR MediaDesign

Tags: acf, animatecss, parameters

Requires at least: 3.9

Tested: WordPress 5.2.2 and ACF 5.8.2

Stable tag: trunk

License: GPLv2 or later

License URI: http://www.gnu.org/licenses/gpl-2.0.html

Forked from: https://github.com/reyhoun/acf-animate-parameter

## Description

Advanced Custom Fields (ACF) field to catch settings for Animate.css like effect to use, animation speed and start delay.

For Animate.css options see https://daneden.github.io/animate.css/ and https://github.com/daneden/animate.css

**Notice:** This Addon does not enqueue Animate.css - yet. Maybe in a later version.

## Compatibility

This ACF field type is compatible with:
* Up to WordPress 5.2.2
* ACF 5 (Tested with 5.8.2)

## Installation

1. Copy the `acf-animatecss` folder into your `wp-content/plugins` folder
2. Activate the `ACF: Animate.css Effects Field` plugin via the plugins admin page
3. Create a new field via ACF and select the Animate Parameters type
4. Please refer to the description for more info regarding the field type settings

CHANGELOG.md
#### 1.2.0
* better ui for user
* fix some language support bug
* fix small bug in html markup
* remove effect name
* add -none- value to select animate type field

#### 1.1.2
* fix version number.


#### 1.1.1
* better markup in admin and user section

#### 1.1.0
* Compatible with [Github Updater](https://github.com/afragen/github-updater)
* add language pot file
* add changelog file
* cleanup readme.md

#### 1.0.0
* Initial Release.

js/input.js
(function($){
	
	
	function initialize_field( $el ) {
		
		//$el.doStuff();
		
	}
	
	
	if( typeof acf.add_action !== 'undefined' ) {
	
		/*
		*  ready append (ACF5)
		*
		*  These are 2 events which are fired during the page load
		*  ready = on page load similar to $(document).ready()
		*  append = on new DOM elements appended via repeater field
		*
		*  @type	event
		*  @date	20/07/13
		*
		*  @param	$el (jQuery selection) the jQuery element which contains the ACF fields
		*  @return	n/a
		*/
		
		acf.add_action('ready append', function( $el ){
			
			// search $el for fields of type 'animate_parameters'
			acf.get_fields({ type : 'animate_parameters'}, $el).each(function(){
				
				initialize_field( $(this) );
				
			});
			
		});
		
		
	} else {
		
		
		/*
		*  acf/setup_fields (ACF4)
		*
		*  This event is triggered when ACF adds any new elements to the DOM. 
		*
		*  @type	function
		*  @since	1.0.0
		*  @date	01/01/12
		*
		*  @param	event		e: an event object. This can be ignored
		*  @param	Element		postbox: An element which contains the new HTML
		*
		*  @return	n/a
		*/
		
		$(document).live('acf/setup_fields', function(e, postbox){
			
			$(postbox).find('.field[data-field_type="animate_parameters"]').each(function(){
				
				initialize_field( $(this) );
				
			});
		
		});
	
	
	}


})(jQuery);


acf-animatecss-v5.php
<?php

class acf_field_animate_parameters extends acf_field {
	
	
	/*
	*  __construct
	*
	*  This function will setup the field type data
	*
	*  @type	function
	*  @date	5/03/2014
	*  @since	5.0.0
	*
	*  @param	n/a
	*  @return	n/a
	*/
	
	function __construct() {
		
		/*
		*  name (string) Single word, no spaces. Underscores allowed
		*/
		
		$this->name = 'acf_animatecss';
		
		
		/*
		*  label (string) Multiple words, can include spaces, visible when selecting a field type
		*/
				$this->label = __('Animate.css Effects', 'acf-animatecss');
				
		/*
		*  category (string) basic | content | choice | relational | jquery | layout | CUSTOM GROUP NAME
		*/
				$this->category = 'basic';
				
		/*
		*  defaults (array) Array of default settings which are merged into the field object. These are used later in settings
		*/		
		$this->defaults = array(
			'effect_name'	=> '',
			'enable_effect'	=> 1,
			'select_animate_type'	=> 1,
			'list_values'			=> array(
										'none' => '-none-',
										'bounce' => 'bounce',
										'flash' => 'flash',
										'pulse' => 'pulse',
										'rubberBand' => 'rubberBand',
										'shake' => 'shake',
										'swing' => 'swing',
										'tada' => 'tada',
										'wobble' => 'wobble',
										'bounceIn' => 'bounceIn',
										'bounceInDown' => 'bounceInDown',
										'bounceInLeft' => 'bounceInLeft',
										'bounceInRight' => 'bounceInRight',
										'bounceInUp' => 'bounceInUp',
										'fadeIn' => 'fadeIn',
										'fadeInDown' => 'fadeInDown',
										'fadeInDownBig' => 'fadeInDownBig',
										'fadeInLeft' => 'fadeInLeft',
										'fadeInLeftBig' => 'fadeInLeftBig',
										'fadeInRight' => 'fadeInRight',
										'fadeInRightBig' => 'fadeInRightBig',
										'fadeInUp' => 'fadeInUp',
										'fadeInUpBig' => 'fadeInUpBig',
										'flip' => 'flip',
										'flipInX' => 'flipInX',
										'flipInY' => 'flipInY',
										'lightSpeedIn' => 'lightSpeedIn',
										'rotateIn' => 'rotateIn',
										'rotateInDownLeft' => 'rotateInDownLeft',
										'rotateInDownRight' => 'rotateInDownRight',
										'rotateInUpLeft' => 'rotateInUpLeft',
										'rotateInUpRight' => 'rotateInUpRight',
										'rollIn' => 'rollIn',
										'zoomIn' => 'zoomIn',
										'zoomInDown' => 'zoomInDown',
										'zoomInLeft' => 'zoomInLeft',
										'zoomInRight' => 'zoomInRight',
										'zoomInUp' => 'zoomInUp',
										'slideInDown' => 'slideInDown',
										'slideInLeft' => 'slideInLeft',
										'slideInRight' => 'slideInRight',
										'slideInUp' => 'slideInUp'
								),
				'delay'	=>	array(
											0	=>	'0',
											1	=>	'1',
											2	=>	'2',
											3	=>	'3',
											4	=>	'4',
											5	=>	'5',
										),
		);
						
		// do not delete!
    	parent::__construct();
    	
	}
	
	
	/*
	*  render_field_settings()
	*
	*  Create extra settings for your field. These are visible when editing a field
	*
	*  @type	action
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$field (array) the $field being edited
	*  @return	n/a
	*/
	
	function render_field_settings( $field ) {
		
		/*
		*  acf_render_field_setting
		*
		*  This function will create a setting for your field. Simply pass the $field parameter and an array of field settings.
		*  The array of settings does not require a `value` or `prefix`; These settings are found from the $field array.
		*
		*  More than one setting can be added by copy/paste the above code.
		*  Please note that you must also have a matching $defaults value for the field name (font_size)
		*/
		
		acf_render_field_setting( $field, array(
			'label'			=> __('Enable Effect','acf-animatecss'),
			'type'			=> 'radio',
			'name'			=> 'enable_effect',
			'layout'  		=>  'horizontal',
			'choices' =>  array(
											1 => __('Yes', 'acf-animatecss'),
											0 => __('No', 'acf-animatecss'),
										)
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Select Animate.css Type','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'select_animate_type',
			'choices'		=> $field['list_values'],
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animate.css Start Delay','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'animate_start_delay',
			'choices' =>  array(
											0	=>	0,
											1	=>	1,
											2	=>	2,
											3	=>	3,
											4	=>	4,
											5	=>	5,
										),
			'append'		=> 'Seconds'
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animation Speed','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'animation_speed',
			'choices' =>  array(
											'slower'	=>	__('Slowest (3s)','acf-animatecss'),
											'slow'		=>	__('Slow (2s)','acf-animatecss'),
											'fast'		=>	__('Fast (800ms)','acf-animatecss'),
											'faster'	=>	__('Fastest (500ms)','acf-animatecss'),
										),
			'append'		=> '',
		));
	}


	/*
	*  render_field()
	*
	*  Create the HTML interface for your field
	*
	*  @param	$field (array) the $field being rendered
	*
	*  @type	action
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$field (array) the $field being edited
	*  @return	n/a
	*/
	function render_field( $field ) {
		
			// add empty value (allows '' to be selected)
			if( empty($field['value']) ){

					$field['value']['enable_effect'] 					= $field['enable_effect'];
					$field['value']['select_animate_type']		= $field['select_animate_type'];
					$field['value']['animate_start_delay'] 		= $field['animate_start_delay'];
					$field['value']['animation_speed']       	= $field['animation_speed'];
			}

			$field_value = $field['value'];

			$e = '<div class="acf-label">';
			$e.= '<label class="" for="' . $field['key'] . '">' . __("Enable Effect", "acf-animate_parameters") . '</label>';
			$e.= '<div class="acf-input">';
			$e.=  '<ul class="acf-radio-list acf-hl"><li><label><input type="radio" name="' . $field['name'] . '[enable_effect]"'; if ($field_value['enable_effect']) { $e.= 'checked'; } $e.= ' value="1">' . __("Yes", "acf-animate_parameters") . '</label></li><li><label>';
			$e.=  '<input type="radio" name="' . $field['name'] . '[enable_effect]"'; if (!$field_value['enable_effect']) { $e.= 'checked'; } $e.= ' value="0">' . __("No", "acf-animate_parameters") . '</label></li></ul>';
			$e.= '</div></div>';


			$e.= '<div class="acf-label">';
			$e.= '<div class="acf-input-prepend">' . __("Animate.css Animation", "acf-animate_parameters") . $i . '</div>';
			$e.= '<div class="acf-input-wrap">';
			$e.= '<select name="' . $field['name'] . '[select_animate_type]" class="js-select2">';
				foreach ( $field['list_values'] as $k => $v ) {
					$e.= '<option value="' . $k . '"' . selected($field_value['select_animate_type'], $k, false) . ' >' . $k . '</option>' ;
				}
			$e.= '</select>';	
			$e.= '</div></div>';


			$e.= '<div class="acf-label">';
			$e.= '<div class="acf-input-prepend">' . __('Animate.css Start Delay','acf-animatecss') . '</div>';
			$e.= '<div class="acf-input-wrap">';
			$e.= '<select name="' . $field['name'] . '[animate_start_delay]" class="js-select2">';
				foreach ( $field['delay'] as $k => $v ) {
					$e.= '<option value="' . $k . '"' . selected($field_value['animate_start_delay'], $k, false) . ' >' . $k . '</option>' ;
				}
			$e.= '</select>';	
			$e.= '</div></div>';


			$e.= '<div class="acf-label">';
			$e.= '<div class="acf-input-prepend">' . __('Animation Speed','acf-animatecss') . '</div>';
			$e.= '<div class="acf-input-wrap">';
			$e.= '<select name="' . $field['name'] . '[animation_speed]" class="js-select2">';
				foreach ( $field['animation_speed_values'] as $k => $v ) {
					$e.= '<option value="' . $k . '"' . selected($field_value['animation_speed'], $k, false) . ' >' . $k . '</option>' ;
				}
			$e.= '</select>';	
			$e.= '</div></div>';
			
			echo $e;
	}

}

// create field
new acf_field_animate_parameters();
?>


acf-animate-parameters.php
<?php

/*
 * Plugin Name: ACF: Animate.css Effects
 * Plugin URI: https://github.com/Thomas-A-Reinert/acf-animatecss
 * Description: ACF field to catch settings for Animate.css like effect to use, animation speed and start delay.
 * Version: 0.1
 * Author: Thomas A. Reinert, TAR MediaDesign
 * Author URI: https://www.tarcgn.de/
 * License: GPLv2 or later
 * License URI: http://www.gnu.org/licenses/gpl-2.0.html
*/




// 1. set text domain
// Reference: https://codex.wordpress.org/Function_Reference/load_plugin_textdomain
load_plugin_textdomain( 'acf-animatecss', false, dirname( plugin_basename(__FILE__) ) . '/lang/' ); 




// 2. Include field type for ACF5
// $version = 5 and can be ignored until ACF6 exists
function include_field_types_animate_parameters( $version ) {
	
	include_once('acf-animatecss-v5.php');
	
}

add_action('acf/include_field_types', 'include_field_types_animate_parameters');	
?>

acf-animate-parameters-v5.php
<?php

class acf_field_animate_parameters extends acf_field {
	
	
	/*
	*  __construct
	*
	*  This function will setup the field type data
	*
	*  @type	function
	*  @date	5/03/2014
	*  @since	5.0.0
	*
	*  @param	n/a
	*  @return	n/a
	*/
	
	function __construct() {
		
		/*
		*  name (string) Single word, no spaces. Underscores allowed
		*/
		
		$this->name = 'animate_parameters';
		
		
		/*
		*  label (string) Multiple words, can include spaces, visible when selecting a field type
		*/
		
		$this->label = __('Animate.css Effects', 'acf-animatecss');
		
		
		/*
		*  category (string) basic | content | choice | relational | jquery | layout | CUSTOM GROUP NAME
		*/
		
		$this->category = 'basic';
		
		
		/*
		*  defaults (array) Array of default settings which are merged into the field object. These are used later in settings
		*/
		
		$this->defaults = array(
			'font_size'		=> 14,
			'effect_name'	=> '',
			'enable_effect'	=> 1,
			'select_animate_type'	=> 1,
			'select_animate_type_1'	=> 'none',
			'select_animate_type_2'	=> 'none',
			'select_animate_type_3'	=> 'none',
			'select_animate_type_4'	=> 'none',
			'animate_delay_time'	=> 10,
			'increase_delay'		=> 10,
			'list_values'			=> array(
										'none' => '-none-',
										'bounce' => 'bounce',
										'flash' => 'flash',
										'pulse' => 'pulse',
										'rubberBand' => 'rubberBand',
										'shake' => 'shake',
										'swing' => 'swing',
										'tada' => 'tada',
										'wobble' => 'wobble',
										'bounceIn' => 'bounceIn',
										'bounceInDown' => 'bounceInDown',
										'bounceInLeft' => 'bounceInLeft',
										'bounceInRight' => 'bounceInRight',
										'bounceInUp' => 'bounceInUp',
										'fadeIn' => 'fadeIn',
										'fadeInDown' => 'fadeInDown',
										'fadeInDownBig' => 'fadeInDownBig',
										'fadeInLeft' => 'fadeInLeft',
										'fadeInLeftBig' => 'fadeInLeftBig',
										'fadeInRight' => 'fadeInRight',
										'fadeInRightBig' => 'fadeInRightBig',
										'fadeInUp' => 'fadeInUp',
										'fadeInUpBig' => 'fadeInUpBig',
										'flip' => 'flip',
										'flipInX' => 'flipInX',
										'flipInY' => 'flipInY',
										'lightSpeedIn' => 'lightSpeedIn',
										'rotateIn' => 'rotateIn',
										'rotateInDownLeft' => 'rotateInDownLeft',
										'rotateInDownRight' => 'rotateInDownRight',
										'rotateInUpLeft' => 'rotateInUpLeft',
										'rotateInUpRight' => 'rotateInUpRight',
										'rollIn' => 'rollIn',
										'zoomIn' => 'zoomIn',
										'zoomInDown' => 'zoomInDown',
										'zoomInLeft' => 'zoomInLeft',
										'zoomInRight' => 'zoomInRight',
										'zoomInUp' => 'zoomInUp',
										'slideInDown' => 'slideInDown',
										'slideInLeft' => 'slideInLeft',
										'slideInRight' => 'slideInRight',
										'slideInUp' => 'slideInUp'
								),
				'delay'	=>	array(
											0	=>	0,
											1	=>	1,
											2	=>	2,
											3	=>	3,
											4	=>	4,
											5	=>	5,
										),
				'animation_speed'	=>	array(
																'slower'	=>	'Slowest (3s)',
																'slow'		=>	'Slow (2s)',
																'fast'		=>	'Fast (800ms)',
																'faster'	=>	'Fastest (500ms)',
															),
		);
		
		
		/*
		*  l10n (array) Array of strings that are used in JavaScript. This allows JS strings to be translated in PHP and loaded via:
		*  var message = acf._e('animate_parameters', 'error');
		*/
		
		$this->l10n = array(
			'error'	=> __('Error! Please enter a higher value', 'acf-animatecss'),
		);
		
				
		// do not delete!
    	parent::__construct();
    	
	}
	
	
	/*
	*  render_field_settings()
	*
	*  Create extra settings for your field. These are visible when editing a field
	*
	*  @type	action
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$field (array) the $field being edited
	*  @return	n/a
	*/
	
	function render_field_settings( $field ) {
		
		/*
		*  acf_render_field_setting
		*
		*  This function will create a setting for your field. Simply pass the $field parameter and an array of field settings.
		*  The array of settings does not require a `value` or `prefix`; These settings are found from the $field array.
		*
		*  More than one setting can be added by copy/paste the above code.
		*  Please note that you must also have a matching $defaults value for the field name (font_size)
		*/
		
		acf_render_field_setting( $field, array(
			'label'			=> __('Enable Effect','acf-animatecss'),
			'type'			=> 'radio',
			'name'			=> 'enable_effect',
			'layout'  		=>  'horizontal',
			'choices' =>  array(
                1 => __('Yes', 'acf-animatecss'),
                0 => __('No', 'acf-animatecss'),
            )
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animate Type 1','acf-animatecss'),
			'instructions'	=> __('','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'select_animate_type_1',
			'choices'		=> $field['list_values'],
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animate Type 2','acf-animatecss'),
			'instructions'	=> __('','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'select_animate_type_2',
			'choices'		=> $field['list_values'],
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animate Type 3','acf-animatecss'),
			'instructions'	=> __('','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'select_animate_type_3',
			'choices'		=> $field['list_values'],
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animate Type 4','acf-animatecss'),
			'instructions'	=> __('','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'select_animate_type_4',
			'choices'		=> $field['list_values'],
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Select Animate Type','acf-animatecss'),
			'instructions'	=> __('Select number of animate select type to show user.','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'select_animate_type',
			'choices' =>  array(
											1	=>	1,
											2	=>	2,
											3	=>	3,
											4	=>	4,
										)
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animate Start Delay','acf-animatecss'),
			'instructions'	=> __('','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'animate_start_delay',
			'choices' =>  array(
											0	=>	0,
											1	=>	1,
											2	=>	2,
											3	=>	3,
											4	=>	4,
											5	=>	5,
										),
			'append'		=> 'Seconds'
		));

		acf_render_field_setting( $field, array(
			'label'			=> __('Animation Speed','acf-animatecss'),
			'instructions'	=> __('','acf-animatecss'),
			'type'			=> 'select',
			'ui'			=> 1,
			'name'			=> 'animation_speed',
			'choices' =>  array(
											'slower'	=>	'Slowest (3s)',
											'slow'		=>	'Slow (2s)',
											'fast'		=>	'Fast (800ms)',
											'faster'	=>	'Fastest (500ms)',
										),
			'append'		=> '',
		));

	}
	
	
	
	/*
	*  render_field()
	*
	*  Create the HTML interface for your field
	*
	*  @param	$field (array) the $field being rendered
	*
	*  @type	action
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$field (array) the $field being edited
	*  @return	n/a
	*/
	
	function render_field( $field ) {
		
		  // add empty value (allows '' to be selected)
        if( empty($field['value']) ){

            $field['value']['enable_effect'] 					= $field['enable_effect'];
            $field['value']['select_animate_type_1'] 	= $field['select_animate_type_1'];
            $field['value']['select_animate_type_2'] 	= $field['select_animate_type_2'];
            $field['value']['select_animate_type_3'] 	= $field['select_animate_type_3'];
            $field['value']['select_animate_type_4'] 	= $field['select_animate_type_4'];
            $field['value']['animate_start_delay'] 		= $field['animate_start_delay'];
            $field['value']['animation_speed']       	= $field['animation_speed'];
        }

        $field_value = $field['value'];
		/*
		*  Review the data of $field.
		*  This will show what data is available
		*/
		
		// echo '<pre>';
		// 	print_r( $field );
		// echo '</pre>';
		

			$e = '<div class="acf-label">';
			$e.= '<label class="" for="' . $field['key'] . '">' . __("Enable Effect", "acf-animate_parameters") . '</label>';
			$e.= '<div class="acf-input">';
			$e.=  '<ul class="acf-radio-list acf-hl"><li><label><input type="radio" name="' . $field['name'] . '[enable_effect]"'; if ($field_value['enable_effect']) { $e.= 'checked'; } $e.= ' value="1">' . __("Yes", "acf-animate_parameters") . '</label></li><li><label>';
			$e.=  '<input type="radio" name="' . $field['name'] . '[enable_effect]"'; if (!$field_value['enable_effect']) { $e.= 'checked'; } $e.= ' value="0">' . __("No", "acf-animate_parameters") . '</label></li></ul>';
			$e.= '</div></div>';


		for ($i=1; $i <= $field['select_animate_type'] ; $i++) {

			$e.= '<div class="acf-label">';
			$e.= '<div class="acf-input-prepend">' . __("Animate #", "acf-animate_parameters") . $i . '</div>';
			$e.= '<div class="acf-input-wrap">';
			$e.= '<select name="' . $field['name'] . '[select_animate_type_'. $i . ']" class="js-select2">';
				foreach ( $field['list_values'] as $k => $v ) {
					$e.= '<option value="' . $k . '"' . selected($field_value['select_animate_type_' . $i], $k, false) . ' >' . $k . '</option>' ;
				}
			$e.= '</select>';	
			$e.= '</div></div>';
		}


			$e.= '<div class="acf-label">';
			$e.= '<div class="acf-input-prepend">' . __('Animate Start Delay','acf-animatecss') . '</div>';
			$e.= '<div class="acf-input-wrap">';
			$e.= '<select name="' . $field['name'] . '[animate_start_delay]" class="js-select2">';
				foreach ( $field['delay'] as $k => $v ) {
					$e.= '<option value="' . $k . '"' . selected($k, false) . ' >' . $k . '</option>' ;
				}
			$e.= '</select>';	
			$e.= '</div></div>';


			$e.= '<div class="acf-label">';
			$e.= '<div class="acf-input-prepend">' . __('Animation Speed','acf-animatecss') . '</div>';
			$e.= '<div class="acf-input-wrap">';
			$e.= '<select name="' . $field['name'] . '[animation_speed]" class="js-select2">';
				foreach ( $field['animation_speed'] as $k => $v ) {
					$e.= '<option value="' . $k . '"' . selected($k, false) . ' >' . $k . '</option>' ;
				}
			$e.= '</select>';	
			$e.= '</div></div>';
			
			echo $e;
	}
	
		
	/*
	*  input_admin_enqueue_scripts()
	*
	*  This action is called in the admin_enqueue_scripts action on the edit screen where your field is created.
	*  Use this action to add CSS + JavaScript to assist your render_field() action.
	*
	*  @type	action (admin_enqueue_scripts)
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	n/a
	*  @return	n/a
	*/

	/*
	
	function input_admin_enqueue_scripts() {
		
		$dir = plugin_dir_url( __FILE__ );
		
		
		// register & include JS
		wp_register_script( 'acf-input-animate_parameters', "{$dir}js/input.js" );
		wp_enqueue_script('acf-input-animate_parameters');
		
		
		// register & include CSS
		wp_register_style( 'acf-input-animate_parameters', "{$dir}css/input.css" ); 
		wp_enqueue_style('acf-input-animate_parameters');
		
		
	}
	
	*/
	
	
	/*
	*  input_admin_head()
	*
	*  This action is called in the admin_head action on the edit screen where your field is created.
	*  Use this action to add CSS and JavaScript to assist your render_field() action.
	*
	*  @type	action (admin_head)
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	n/a
	*  @return	n/a
	*/

	/*
		
	function input_admin_head() {
	
		
		
	}
	
	*/
	
	
	/*
   	*  input_form_data()
   	*
   	*  This function is called once on the 'input' page between the head and footer
   	*  There are 2 situations where ACF did not load during the 'acf/input_admin_enqueue_scripts' and 
   	*  'acf/input_admin_head' actions because ACF did not know it was going to be used. These situations are
   	*  seen on comments / user edit forms on the front end. This function will always be called, and includes
   	*  $args that related to the current screen such as $args['post_id']
   	*
   	*  @type	function
   	*  @date	6/03/2014
   	*  @since	5.0.0
   	*
   	*  @param	$args (array)
   	*  @return	n/a
   	*/
   	
   	/*
   	
   	function input_form_data( $args ) {
	   	
		
	
   	}
   	
   	*/
	
	
	/*
	*  input_admin_footer()
	*
	*  This action is called in the admin_footer action on the edit screen where your field is created.
	*  Use this action to add CSS and JavaScript to assist your render_field() action.
	*
	*  @type	action (admin_footer)
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	n/a
	*  @return	n/a
	*/

	/*
		
	function input_admin_footer() {
	
		
		
	}
	
	*/
	
	
	/*
	*  field_group_admin_enqueue_scripts()
	*
	*  This action is called in the admin_enqueue_scripts action on the edit screen where your field is edited.
	*  Use this action to add CSS + JavaScript to assist your render_field_options() action.
	*
	*  @type	action (admin_enqueue_scripts)
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	n/a
	*  @return	n/a
	*/

	/*
	
	function field_group_admin_enqueue_scripts() {
		
	}
	
	*/

	
	/*
	*  field_group_admin_head()
	*
	*  This action is called in the admin_head action on the edit screen where your field is edited.
	*  Use this action to add CSS and JavaScript to assist your render_field_options() action.
	*
	*  @type	action (admin_head)
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	n/a
	*  @return	n/a
	*/

	/*
	
	function field_group_admin_head() {
	
	}
	
	*/


	/*
	*  load_value()
	*
	*  This filter is applied to the $value after it is loaded from the db
	*
	*  @type	filter
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$value (mixed) the value found in the database
	*  @param	$post_id (mixed) the $post_id from which the value was loaded
	*  @param	$field (array) the field array holding all the field options
	*  @return	$value
	*/
	
	/*
	
	function load_value( $value, $post_id, $field ) {
		
		return $value;
		
	}
	
	*/
	
	
	/*
	*  update_value()
	*
	*  This filter is applied to the $value before it is saved in the db
	*
	*  @type	filter
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$value (mixed) the value found in the database
	*  @param	$post_id (mixed) the $post_id from which the value was loaded
	*  @param	$field (array) the field array holding all the field options
	*  @return	$value
	*/
	
	/*
	
	function update_value( $value, $post_id, $field ) {
		
		return $value;
		
	}
	
	*/
	
	
	/*
	*  format_value()
	*
	*  This filter is appied to the $value after it is loaded from the db and before it is returned to the template
	*
	*  @type	filter
	*  @since	3.6
	*  @date	23/01/13
	*
	*  @param	$value (mixed) the value which was loaded from the database
	*  @param	$post_id (mixed) the $post_id from which the value was loaded
	*  @param	$field (array) the field array holding all the field options
	*
	*  @return	$value (mixed) the modified value
	*/
		
	/*
	
	function format_value( $value, $post_id, $field ) {
		
		// bail early if no value
		if( empty($value) ) {
		
			return $value;
			
		}
		
		
		// apply setting
		if( $field['font_size'] > 12 ) { 
			
			// format the value
			// $value = 'something';
		
		}
		
		
		// return
		return $value;
	}
	
	*/
	
	
	/*
	*  validate_value()
	*
	*  This filter is used to perform validation on the value prior to saving.
	*  All values are validated regardless of the field's required setting. This allows you to validate and return
	*  messages to the user if the value is not correct
	*
	*  @type	filter
	*  @date	11/02/2014
	*  @since	5.0.0
	*
	*  @param	$valid (boolean) validation status based on the value and the field's required setting
	*  @param	$value (mixed) the $_POST value
	*  @param	$field (array) the field array holding all the field options
	*  @param	$input (string) the corresponding input name for $_POST value
	*  @return	$valid
	*/
	
	/*
	
	function validate_value( $valid, $value, $field, $input ){
		
		// Basic usage
		if( $value < $field['custom_minimum_setting'] )
		{
			$valid = false;
		}
		
		
		// Advanced usage
		if( $value < $field['custom_minimum_setting'] )
		{
			$valid = __('The value is too little!','acf-animatecss'),
		}
		
		
		// return
		return $valid;
		
	}
	
	*/
	
	
	/*
	*  delete_value()
	*
	*  This action is fired after a value has been deleted from the db.
	*  Please note that saving a blank value is treated as an update, not a delete
	*
	*  @type	action
	*  @date	6/03/2014
	*  @since	5.0.0
	*
	*  @param	$post_id (mixed) the $post_id from which the value was deleted
	*  @param	$key (string) the $meta_key which the value was deleted
	*  @return	n/a
	*/
	
	/*
	
	function delete_value( $post_id, $key ) {
		
		
		
	}
	
	*/
	
	
	/*
	*  load_field()
	*
	*  This filter is applied to the $field after it is loaded from the database
	*
	*  @type	filter
	*  @date	23/01/2013
	*  @since	3.6.0	
	*
	*  @param	$field (array) the field array holding all the field options
	*  @return	$field
	*/
	
	/*
	
	function load_field( $field ) {
		
		return $field;
		
	}	
	
	*/
	
	
	/*
	*  update_field()
	*
	*  This filter is applied to the $field before it is saved to the database
	*
	*  @type	filter
	*  @date	23/01/2013
	*  @since	3.6.0
	*
	*  @param	$field (array) the field array holding all the field options
	*  @return	$field
	*/
	
	/*
	
	function update_field( $field ) {
		
		return $field;
		
	}	
	
	*/
	
	
	/*
	*  delete_field()
	*
	*  This action is fired after a field is deleted from the database
	*
	*  @type	action
	*  @date	11/02/2014
	*  @since	5.0.0
	*
	*  @param	$field (array) the field array holding all the field options
	*  @return	n/a
	*/
	
	/*
	
	function delete_field( $field ) {
		
		
		
	}	
	
	*/
	
	
}


// create field
new acf_field_animate_parameters();

?>
